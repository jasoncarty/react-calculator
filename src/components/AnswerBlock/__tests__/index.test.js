import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import { AnswerBlock } from "../index";

describe("<AnswerBlock />", () => {
  it("matches snapshot", () => {
    const instance = ReactTestRenderer.create(
      <AnswerBlock
        fieldValues={{ 0: "2", 1: "6", 2: "20" }}
        updateCalculationType={() => {}}
        calculationType={""}
      />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("matches snapshot", () => {
    const instance = ReactTestRenderer.create(
      <AnswerBlock
        fieldValues={{ 0: "2", 1: "6", 2: "20" }}
        updateCalculationType={() => {}}
        calculationType={"*"}
      />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("matches snapshot", () => {
    const instance = ReactTestRenderer.create(
      <AnswerBlock
        fieldValues={{ 0: "2", 1: "6", 2: "20" }}
        updateCalculationType={() => {}}
        calculationType={"+"}
      />
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });

  it("updates state", () => {
    const instance = ReactTestRenderer.create(
      <AnswerBlock
        fieldValues={{ 0: "2", 1: "6", 2: "20" }}
        updateCalculationType={() => {}}
        calculationType={"+"}
      />
    );

    instance.update(
      <AnswerBlock
        fieldValues={{ 0: "4", 1: "6", 2: "20" }}
        updateCalculationType={() => {}}
        calculationType={"+"}
      />
    );

    expect(instance.getInstance().state)
      .toEqual({
        answer: 30
      });
  });

  it("calls updateCalculationType", () => {
    const mockFn = jest.fn();
    const instance = ReactTestRenderer.create(
      <AnswerBlock
        fieldValues={{ 0: "2", 1: "6", 2: "20" }}
        updateCalculationType={mockFn}
        calculationType={"+"}
      />
    );

    const sumRadio = instance.root.findByProps({ value: "SUM"});
    sumRadio.props.onChange({
      target: {
        value: "mockValue"
      }
    });

    expect(mockFn).toBeCalledWith("mockValue");
  });
});
