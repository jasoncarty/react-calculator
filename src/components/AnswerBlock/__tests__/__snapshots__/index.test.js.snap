// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnswerBlock /> matches snapshot 1`] = `
<div
  className="answerBlock"
>
  <div
    className="content"
  >
    <div
      className="radioList"
    >
      <input
        checked={false}
        name="calculationType"
        onChange={[Function]}
        type="radio"
        value="SUM"
      />
      Add
      <input
        checked={false}
        name="calculationType"
        onChange={[Function]}
        type="radio"
        value="MULTIPLY"
      />
      Multiply
    </div>
    <div
      className="answer"
    >
      Answer: 
      0
    </div>
  </div>
</div>
`;

exports[`<AnswerBlock /> matches snapshot 2`] = `
<div
  className="answerBlock"
>
  <div
    className="content"
  >
    <div
      className="radioList"
    >
      <input
        checked={false}
        name="calculationType"
        onChange={[Function]}
        type="radio"
        value="SUM"
      />
      Add
      <input
        checked={true}
        name="calculationType"
        onChange={[Function]}
        type="radio"
        value="MULTIPLY"
      />
      Multiply
    </div>
    <div
      className="answer"
    >
      Answer: 
      240
    </div>
  </div>
</div>
`;

exports[`<AnswerBlock /> matches snapshot 3`] = `
<div
  className="answerBlock"
>
  <div
    className="content"
  >
    <div
      className="radioList"
    >
      <input
        checked={true}
        name="calculationType"
        onChange={[Function]}
        type="radio"
        value="SUM"
      />
      Add
      <input
        checked={false}
        name="calculationType"
        onChange={[Function]}
        type="radio"
        value="MULTIPLY"
      />
      Multiply
    </div>
    <div
      className="answer"
    >
      Answer: 
      28
    </div>
  </div>
</div>
`;
